bs/src/tablecloth.mli

```diff
@@ -1527,41 +1527,345 @@ module Result : sig
     -> unit
 end
 
+(**
+  This module implements the `Option` type, which has a variant for 
+  valid values (`'Some`), and one for invalid values (`'None`).
+*)
+
 module Option : sig
+  (**
+    `type` is the type constructor for an `Option` type. You specify
+    the type of the `Some` variant.
+    
+    ### Example
+    
+    Here is how you would annotate an `Option` variable whose `Some`
+    variant is an integer:
+    
+    ```ocaml
+    let x: (int) Tablecloth.Option.t = Some 3
+    let y: (int) Tablecloth.Option.t = None
+    ```
+    
+    ```reason
+    let x: Tablecloth.Result.t(string, int) = Ok(3);
+    let y: Tablecloth.Result.t(string, int) = Error("bad");
+    ```
+  *)
   type 'a t = 'a option
 
+  (**
+    `andThen(~f = fcn, opt)` applies function `fcn`, which takes a non-`Option`
+    parameter and returns a `Option`, to an `Option` variable `opt`.
+    
+    If `opt` is of the form `Some(x)`, `andThen` returns `f(x)`;
+    otherwise `andThen` returns `None`.
+    
+    (Same as `and_then`.)
+    
+    ### Example
+    ```reason
+    let recip = (x: float): Tablecloth.Option.t(float) => {
```

Comment:
lets use the built-in option type in these examples.

Reply:
OK will do.